### PR TITLE
Add site level follow ups

### LINKED ISSUE #56: Add site level follow ups
Some site level screenings may occur as a follow-up to another site level screening with a higher limit on detection resolution. For example satellite -> plane -> OGI FU. This capability needs to be added to LDAR-Sim.

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
Visit site logic was altered to support unflagging/reflagging of sites when the measurement scale of the sensor used by the crew visiting the site is "site-level" and the crew belongs to a follow-up method. Unit testing was added to verify functionality of the modified visit_site functionality.

This change was made to resolve the need for some site-level measurement scale methods method to be follow-up methods, as described by #56.

The purpose of this changes is to enable the use of site-level measurement scale follow-up method in LDAR-Sim.

This pull request is anticipated to result in a patch level version bump from 2.1.3 to 2.1.4

Closes #56 

E2E testing passed: 
[E2E testing Results.zip](https://github.com/LDAR-Sim/LDAR_Sim/files/11793885/E2E.testing.Results.zip)

### CODE REVIEW

**Review by SEAJang — APPROVED** [review]
Looks good, though maybe update the wording in the user manual before merge.

**Review by SEAJang — APPROVED** [review]

**Review by KeeganShaw-GIS — CHANGES_REQUESTED** [review]
Also improve documentation to illustrate a use case for this.

**Comment by ThomasGalesloot** [comment]
> Also improve documentation to illustrate a use case for this.

The documentation has been updated to better Illustrate why this might be useful

File: USER_MANUAL.md [hunk 1]
```diff
@@ -1290,11 +1290,11 @@ Method costs. Currency is not important but must be consistent across all inputs
 
 **Default input:** N/A
 
-**Description:** Allows for surveying methods to determine which follow-up method to run if multiples are present. If not set, LDAR-Sim will expect only 1 followup method to be present for a single program.
+**Description:** Allows for surveying methods to determine which follow-up method to run if multiples are present. If not set, LDAR-Sim will expect only 1 followup method to be present for a single program. Site level follow-up methods can be used by leveraging this parameter, allowing for a subsequent rounds of screenings based on initial screening results (In order to increase confidence in fugitives, measurement accuracy, etc). The site level follow-up must be set as the preferred_method for the initial screening.
```
Comment (SEAJang) [inline USER_MANUAL.md:1293]: Please remove the 'a' between "allowing for a subsequent..." 
I would also remove the brackets around "In order to..." and just have it as a sentence. Maybe reword it by saying "This feature can be used to increase confidence... " 
Suggestion in rewording the last sentence: Each method in the follow-up chain must have the preferred_method set to the corresponding method it triggers.

**Review by SEAJang — APPROVED** [review]
Great!
